src/main/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAO.java

```diff
@@ -326,19 +327,32 @@ INSERT INTO dataset_property (dataset_id, property_key, schema_property, propert
   private List<Update> generatePropertyUpdates(Handle handle, Integer datasetId,
       List<DatasetProperty> properties, Set<DatasetProperty> existingProps) {
     List<Update> updates = new ArrayList<>();
-    HashSet<String> existingPropNames = new HashSet<>(
-        existingProps.stream().map(DatasetProperty::getPropertyName).toList());
     // Generate value updates for props that exist
     properties.forEach(prop -> {
-      if (existingPropNames.contains(prop.getPropertyName())) {
-        prop.setDataSetId(datasetId);
-        updates.add(createPropertyUpdate(handle, prop));
+      List<DatasetProperty> matchingProps = existingProps
+          .stream()
+          .filter(ep -> ep.getPropertyName().equals(prop.getPropertyName()))
+          .toList();
+      if (!matchingProps.isEmpty()) {
```

Comment:
nit: is this `if` check required? If there are no items in the list, the size will be 0 and the `forEach` will be skipped.